app/assets/scripts/views/home.js

```diff
@@ -0,0 +1,266 @@
+        'sensor': Joi.string().required().label('Sensor'),
+        'date-start': Joi.date().required().label('Date start'),
+        // 'date-end': non required,
+        'urls': Joi.string().required().label('Imagery location'),
+        'tile-url': Joi.string().required().label('Tile service'),
+        'provider': Joi.string().required().label('Provider'),
+        'contact-type': Joi.string().required().valid('uploader', 'other'),
+        'contact-name': Joi.string().allow('').label('Name'),
+        'contact-email': Joi.label('Email').when('contact-type', { is: 'other', then: Joi.string().email().required() })
+      })
+    )
+  },
+
+  getInitialState: function() {
+    return {
+      loading: false,
+      feedback: {
+        type: null,
+        message: null
+      },
+
+      // Form properties.
+      'uploader-token': '',
+      'uploader-name': '',
+      'uploader-email': '',
+      scenes: [
+        this.getSceneDataTemplate()
+      ]
+    };
+  },
+
+  getSceneDataTemplate: function() {
+    return {
+      'platform-type': 'satellite',
+      'sensor': '',
+      'date-start': new Date().toISOString(),
+      'date-end': null,
+      'urls': '',
+      'tile-url': '',
+      'provider': '',
+      'contact-type': 'uploader',
+      'contact-name': '',
+      'contact-email': ''
+    };
+  },
+
+  addScene: function() {
+    var scenes = this.state.scenes;
+    scenes.push(this.getSceneDataTemplate());
+    this.setState({scenes: scenes});
+  },
+
+  removeScene: function(sceneIndex) {
+    var scenes = this.state.scenes;
+    scenes.splice(sceneIndex, 1);
+    this.setState({scenes: scenes});
+  },
+
+  onSceneValueChange: function(sceneIndex, fieldName, fieldValue) {
+    var scenes = this.state.scenes;
+    scenes[sceneIndex][fieldName] = fieldValue;
+    this.setState({scenes: scenes});
+  },
+
+  onValueChange: function(event) {
+    var data = {};
+    data[event.target.name] = event.target.value;
+    this.setState(data);
+  },
+
+  resetForm: function() {
+    this.setState({
+      'uploader-token': null,
+      'uploader-name': null,
+      'uploader-email': null,
+      scenes: [
+        this.getSceneDataTemplate()
+      ]
+    });
+  },
+
+  onSubmit: function(event) {
+    event.preventDefault();
+    
+    // Warning... Controlled HACK.
+    // The state should never be changed in this way as it doesn't trigger
+    // a render, however it will be updated by the validate function later on.
+    // This is needed to clear previous errors as the plugin doesn't handle
+    // arrays of objects specially well.
+    this.state.errors = {};
+
+    this.validate(function(error, validationErrors) {
+      if (error) {
+        console.log(validationErrors);
+      } else {
+
+        if (this.state.loading) {
+          // Submit already in process.
+          return;
+        }
+        this.setState({loading: true});
+
+        // All is well.
+        // SUBMIT DATA.
```

Comment:
Left it ready for you to implement the data submission.
Add the store and actions as you see fit.

@anandthakker